static/js/modules/election-lookup.js

```diff
@@ -214,16 +216,27 @@ ElectionLookup.prototype.search = function(e, opts) {
   opts = _.extend({pushState: true}, opts || {});
   var self = this;
   var serialized = self.serialize();
-  if (self.shouldSearch(serialized) && !_.isEqual(serialized, self.serialized)) {
-    $.getJSON(self.getUrl(serialized)).done(function(response) {
-      // Note: Update district color map before rendering results
-      self.drawDistricts(response.results);
-      self.draw(response.results);
-    });
-    self.serialized = serialized;
-    if (opts.pushState) {
-      window.history.pushState(serialized, null, URI('').query(serialized).toString());
-      analytics.pageView();
+  if (self.shouldSearch(serialized)) {
+    if (!_.isEqual(serialized, self.serialized)) {
+      // Requested search options differ from saved options; request new data.
+      self.xhr && self.xhr.abort();
+      self.xhr = $.getJSON(self.getUrl(serialized)).done(function(response) {
```

Comment:
Good point, but maybe beyond the scope of this patch. I'll add this to https://github.com/18F/openFEC-web-app/issues/733--this may be a good one to work on when I get around to refactoring the data table code.